Source/WebCore/svg/SVGTransformable.cpp

```diff
@@ -99,67 +99,126 @@ template<typename CharacterType> static int parseTransformParamList(StringParsin
 static constexpr std::array requiredValuesForType { 0, 6, 1, 1, 1, 1, 1 };
 static constexpr std::array optionalValuesForType { 0, 0, 1, 1, 2, 0, 0 };
 
-template<typename CharacterType> static std::optional<SVGTransformValue> parseTransformValueGeneric(SVGTransformValue::SVGTransformType type, StringParsingBuffer<CharacterType>& buffer)
+template<typename CharacterType> static RefPtr<SVGTransform> parseTransformGeneric(SVGTransformValue::SVGTransformType type, StringParsingBuffer<CharacterType>& buffer, RefPtr<SVGTransform> reusableValue)
 {
     if (type == SVGTransformValue::SVG_TRANSFORM_UNKNOWN)
-        return std::nullopt;
+        return nullptr;
 
     int valueCount = 0;
     std::array<float, 6> values { 0, 0, 0, 0, 0, 0 };
     if ((valueCount = parseTransformParamList(buffer, values, requiredValuesForType[type], optionalValuesForType[type])) < 0)
-        return std::nullopt;
+        return nullptr;
 
     switch (type) {
     case SVGTransformValue::SVG_TRANSFORM_UNKNOWN:
         ASSERT_NOT_REACHED();
-        return std::nullopt;
+        return nullptr;
 
     case SVGTransformValue::SVG_TRANSFORM_SKEWX: {
+        if (reusableValue) {
+            ASSERT(reusableValue->value().type() == SVGTransformValue::SVG_TRANSFORM_SKEWX);
```

Comment:
Instead of adding ASSERT to each switch case, we can add one ASSERT at the beginning
```
ASSERT_IMPLIES(reusableValue, type == reusableValue->value().type());
```